src/rfq/protocols/hashflow/client.rs

```diff
@@ -433,35 +431,68 @@ impl RFQClient for HashflowClient {
+                        quote_attributes.insert("effective_trader".to_string(), effective_trader);
+                    }
+                    quote_attributes.insert("base_token".to_string(), quote.quote_data.base_token);
+                    quote_attributes
+                        .insert("quote_token".to_string(), quote.quote_data.quote_token);
                     quote_attributes.insert(
-                        "nonce".to_string(),
+                        "base_token_amount".to_string(),
                         Bytes::from(
-                            quote
-                                .quote_data
-                                .nonce
-                                .to_be_bytes()
-                                .to_vec(),
+                            biguint_to_u256(
+                                &BigUint::from_str(&quote.quote_data.base_token_amount).map_err(
+                                    |_| {
+                                        RFQError::ParsingError(format!(
+                                            "Failed to parse base token amount: {}",
+                                            quote.quote_data.base_token_amount
+                                        ))
+                                    },
+                                )?,
+                            )
+                            .to_be_bytes::<32>()
+                            .to_vec(),
+                        ),
+                    );
+                    quote_attributes.insert(
+                        "quote_token_amount".to_string(),
+                        Bytes::from(
+                            biguint_to_u256(
+                                &BigUint::from_str(&quote.quote_data.quote_token_amount).map_err(
+                                    |_| {
+                                        RFQError::ParsingError(format!(
+                                            "Failed to parse quote token amount: {}",
+                                            quote.quote_data.quote_token_amount
+                                        ))
+                                    },
+                                )?,
+                            )
+                            .to_be_bytes::<32>()
+                            .to_vec(),
                         ),
                     );
-                    quote_attributes.insert("tx_id".to_string(), quote.quote_data.tx_id);
-                    quote_attributes.insert("signature".to_string(), quote.signature);
                     quote_attributes.insert(
                         "quote_expiry".to_string(),
                         Bytes::from(
-                            quote
-                                .quote_data
-                                .quote_expiry
-                                .to_be_bytes()
+                            biguint_to_u256(&BigUint::from(quote.quote_data.quote_expiry))
```

Comment:
would it work to define `HashflowQuoteData.quote_expiry` as a U256? because right now it is a u64. And we are converting to BigUint to convert to U256 to convert to Bytes 😬 at least the BigUint step can be removed for sure